AndroidSDKCore/src/main/java/com/leanplum/internal/SocketIOClient.java

```diff
@@ -78,15 +77,17 @@ private static String userAgentString() {
         + Constants.LEANPLUM_VERSION + "/" + Constants.LEANPLUM_PACKAGE_IDENTIFIER + ")";
   }
 
-  private static String downloadUriAsString(final HttpUriRequest req)
+  private static String downloadUriAsString(final String urlString)
       throws IOException {
-    AndroidHttpClient client = AndroidHttpClient.newInstance(userAgentString());
-    try {
-      HttpResponse res = client.execute(req);
-      return readToEnd(res.getEntity().getContent());
-    } finally {
-      client.close();
+    String fullString = "";
```

Comment:
Just a suggestion, but your approach is fine as well with "+" operator since it internally uses StringBuilder.
Instead of String fullString = ""; 

You could also use StringBuilder.

StringBuilder result = new StringBuilder();
String line;
while((line = reader.readLine()) != null) {
    result.append(line);
}
System.out.println(result.toString());